examples/js/loaders/deprecated/LegacyGLTFLoader.js

```diff
@@ -731,28 +735,64 @@ THREE.LegacyGLTFLoader = ( function () {
 					shaderText = shaderText.replace( regEx, 'uv' );
 					break;
 
-				case 'TEXCOORD_1':
-
-					shaderText = shaderText.replace( regEx, 'uv2' );
-					break;
-
 				case 'COLOR_0':
 				case 'COLOR0':
 				case 'COLOR':
 
 					shaderText = shaderText.replace( regEx, 'color' );
 					break;
 
+				case "WEIGHT_0":
+				case "WEIGHT0":
 				case "WEIGHT":
 
 					shaderText = shaderText.replace( regEx, 'skinWeight' );
 					break;
 
+				case "JOINT_0":
+				case "JOINT0":
 				case "JOINT":
 
 					shaderText = shaderText.replace( regEx, 'skinIndex' );
 					break;
 
+				default:
+
+					var underscoreIndex = semantic.indexOf( '_' );
+
+					if ( underscoreIndex !== - 1 ) {
+
+						var attrName = semantic.slice( 0, underscoreIndex ).toLowerCase();
+						var attrNum = parseInt( semantic.slice( underscoreIndex + 1 ) ) + 1;
+
+						switch ( attrName ) {
+
+							case 'position':
+							case 'normal':
+							case 'color':
+
+								shaderText = shaderText.replace( regEx, attrName + attrNum );
+								break;
+
+							case 'texcoord':
+
+								shaderText = shaderText.replace( regEx, 'uv' + attrNum );
+								break;
+
+							case 'weight':
+
+								shaderText = shaderText.replace( regEx, 'skinWeight' + attrNum );
+								break;
+
+							case 'joint':
+
+								shaderText = shaderText.replace( regEx, 'skinIndex' + attrNum );
+								break;
+
+						}
+
+					}
+
```

Comment:
I think this particular `default: ` block can be removed, actually — it is not necessary to rename attributes that three.js does not make direct use of. More explanation below.